app/controllers/ReportProblemController.scala

```diff
@@ -130,48 +130,65 @@ class ReportProblemController @Inject() (
 
   implicit def lang(implicit request: Request[_]): Lang = request.lang
 
-  def index(service: Option[String]) = Action { implicit request =>
-    Ok(page(ReportProblemFormBind.emptyForm(service = service), service))
+  def index(service: Option[String], referrerUrl: Option[String]) = Action { implicit request =>
+    val referrer = referrerUrl.orElse(request.headers.get(REFERER))
+    Ok(page(ReportProblemFormBind.emptyForm(service, referrer), service, referrerUrl))
+  }
+
+  def indexDeprecated(service: Option[String]) = Action { implicit request =>
+    val referrer = request.headers.get(REFERER)
```

Comment:
shouldn't we be getting the referrerUrl on the deprecated route too?

Reply:
Mea culpa - hadn't included that commit in the PR, pushed up now.